tests/unit/ShareStructure.spec.ts

```diff
@@ -0,0 +1,393 @@
+import Vue from 'vue'
+import Vuetify from 'vuetify'
+import { mount, Wrapper, createLocalVue } from '@vue/test-utils'
+import flushPromises from 'flush-promises'
+
+import { ShareStructure } from '@/components/CreateShareStructure'
+import { ShareClassIF } from '@/interfaces'
+
+// Store
+import { getVuexStore } from '@/store'
+
+Vue.use(Vuetify)
+
+let vuetify = new Vuetify({})
+const store = getVuexStore()
+
+// Events
+const addEditShareClassEvent: string = 'addEditClass'
+const addEditShareSeriesEvent: string = 'addEditSeries'
+const removeClassEvent: string = 'removeClass'
+const removeSeriesEvent: string = 'removeSeries'
+const formResetEvent: string = 'resetEvent'
+
+// Input field selectors to test changes to the DOM elements.
+const nameSelector: string = '#txt-name'
+const txtMaxShares: string = '#txt-max-shares'
+const classParValue: string = '#class-par-value'
+const seriesParValue: string = '#series-par-value'
+const seriesCurrency: string = '#series-currency'
+const noParValueSelector: string = '#radio-no-par'
+const parValueSelector: string = '#radio-par-value'
+const specialRightsChkBoxSelector: string = '#special-rights-check-box'
+const doneButtonSelector: string = '#btn-done'
+const removeButtonSelector: string = '#btn-remove'
+const cancelButtonSelector: string = '#btn-cancel'
+const formSelector: string = '.share-structure-form'
+
+/**
+ * Returns the last event for a given name, to be used for testing event propagation in response to component changes.
+ *
+ * @param wrapper the wrapper for the component that is being tested.
+ * @param name the name of the event that is to be returned.
+ *
+ * @returns the value of the last named event for the wrapper.
+ */
+function getLastEvent (wrapper: Wrapper<ShareStructure>, name: string): any {
```

Comment:
Instead of this helper method, you can also do this:
`expect(wrapper.emitted('close').pop()).toEqual([false])`

Reply:
This method is not used actually. Removed it.